stop_and_wait_plugin/src/stop_and_wait_plugin.cpp

```diff
@@ -0,0 +1,386 @@
+            {
+                delta_time = min_timestep_;
+                curr_time = 0.0;
+                //wait
+                lanelet::BasicPoint2d curr_pose (pose_msg_.pose.position.x,pose_msg_.pose.position.y);
+                while(curr_time < minimal_trajectory_duration_)
+                {
+                    PointSpeedPair pair;
+                    pair.point = curr_pose;
+                    pair.speed = 0.0;
+                    points_and_target_speeds.push_back(pair);
+                    curr_time += delta_time;
+
+                    points_and_target_speeds.push_back(pair);
+                }
+            }
+            else
+            {
+                double jerk_req = (2*current_speed_)/pow(maneuver_time_,2);
+                double start_speed = current_speed_;    //Get static value of current speed at start of planning
+                if(jerk_req > max_jerk_limit_)
+                {
+                    //unsafe to stop at the required jerk - reset to max_jerk and go beyond the maneuver end_dist
+                    jerk_ = max_jerk_limit_;  
+                    maneuver_time_ = pow(2*start_speed, 0.5);
+                    double travel_dist_new = start_speed * maneuver_time_ - (0.167 * jerk_ * pow(maneuver_time_,3));
+                    ending_downtrack = travel_dist_new + starting_downtrack;
+                }
+                else jerk_ = jerk_req;
+
+                //get all the lanelets in between starting and ending downtrack on shortest path
+                auto lanelets = wm_->getLaneletsBetween(starting_downtrack, ending_downtrack, true);
+                //record all the lanelets to be added to path
+                std::vector<lanelet::ConstLanelet> lanelets_to_add;
+                for (auto l : lanelets)
+                {
+                    if(visited_lanelets.find(l.id()) == visited_lanelets.end())
+                    {
+                        lanelets_to_add.push_back(l);
+                        visited_lanelets.insert(l.id());
+                    }
+                }
+
+                lanelet::BasicLineString2d route_geometry = carma_wm::geometry::concatenate_lanelets(lanelets_to_add);
+                
+                int points_count = route_geometry.size();
+                delta_time = maneuver_time_/(points_count-1);
+
+                first = true;
+                curr_time = 0.0;
+
+                for(auto p : route_geometry)
+                {
+                    if (first && points_and_target_speeds.size() != 0)
+                    {
+                        first = false;
+
+                        continue; //Skip the first point to avoid duplicates from previous maneuver
+                    }
+                    PointSpeedPair pair;
+                    pair.point = p;
+                    pair.speed = start_speed - (0.5 * jerk_ * pow(curr_time,2));
+
+                    if(p == route_geometry.back()) pair.speed = 0.0;    //force speed to 0 at last point
```

Comment:
Can't say I'm generally a fan of omitting {} for if statements like this, it makes it substantially less clear if line 232 is supposed to be part of that if-clause, or not. Doesn't appear to be a defect here, but implicit curly braces have always been a personal pet-peeve of mine ever since the large Apple SSL bug from 2014.